### PR TITLE
Fix degree completed bug

### PR DESCRIPTION
**Pull request by JonF1**
## Context
It was possible to complete the degree section with only foundation degrees by marking the section as complete and then deleting the non foundation degree

## Changes proposed in this pull request
* Clears degrees completed db field if there are only foundation degrees, this clears radio button on completed section component and sets tag to incomplete on application review page

## Guidance to review
Verify that it is no longer possible to have complete degree section with only foundation degrees

## Link to Trello card

https://trello.com/c/CSRmIpZH/4688-degree-completed-bug

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Review by avinhurry — APPROVED** [review]
Looks good and tested locally, all works. Nice work 🚀

**Review by tomas-stefano — APPROVED** [review]

File: app/controllers/candidate_interface/degrees/degree/review_controller.rb [hunk 1]
```diff
@@ -34,6 +35,20 @@ def application_form_params
...
+        def set_completed_if_only_foundation_degrees
+          if only_foundation_degrees?
+            current_application.update!(degrees_completed: nil)
+          end
+        end
+
+        def only_foundation_degrees?
+          degree_type = []
+          current_application.application_qualifications.degrees.pluck(:qualification_type).map do |degree|
+            degree_type << Hesa::DegreeType&.find_by_name(degree)&.level
```
Comment (stevehook) [inline app/controllers/candidate_interface/degrees/degree/review_controller.rb:48]: As you are using map I don't think you need to build the `degree_type` array manually. e.g.

```
degree_types = current_application.application_qualifications.degrees.pluck(:qualification_type).map do |degree|
  Hesa::DegreeType&.find_by_name(degree)&.level
end
```

**Review by stevehook — APPROVED** [review]
Looks good. One v minor comment inline.

File: app/controllers/candidate_interface/degrees/degree/review_controller.rb (hunk 1 above)
Comment (JonF1) [reply app/controllers/candidate_interface/degrees/degree/review_controller.rb]: Thanks Steve, I've fixed !

**Pull request merged**